src/sync.js

```diff
@@ -610,6 +617,9 @@ SyncApi.prototype._sync = function(syncOptions) {
         client.store.save();
 
         self._sync(syncOptions);
+
+        // Reset after a successful sync
+        failedSyncCount = 0;
```

Comment:
This wants to be within the _sync function in the `done()` (somewhere around `client.store.setSyncToken(data.next_batch);`). This doesn't do what you think because _sync doesn't block, ie. this will set `failedSyncCount` to 0 right after *starting* a sync request.